packages/anoncreds/src/protocols/credentials/v1/V1CredentialProtocol.ts

```diff
@@ -731,11 +732,7 @@ export class V1CredentialProtocol
 
     agentContext.config.logger.debug(`Processing credential request with id ${requestMessage.id}`)
 
-    const credentialRecord = await this.getByThreadAndConnectionId(
-      messageContext.agentContext,
-      requestMessage.threadId,
-      connection?.id
-    )
+    const credentialRecord = await this.getByThreadAndConnectionId(messageContext.agentContext, requestMessage.threadId)
```

Comment:
Any specific reason why the `connection?.id` is removed?

Reply:
Because if we create an offer (not send) it is not linked to a connection. Then if we use it in an oob message, it can create a connection, before the user send a credential request. Then the incoming message has a connection associated, but the credential record doesn't have.

So we now fetch the credential record without connectionId, and later we check whether the connectionId matches, or if the credential record has no connectionID, and then we need a matching out of band record.